src/modules/aws.rs

```diff
@@ -1,16 +1,85 @@
 use std::env;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use dirs::home_dir;
 
 use super::{Context, Module, RootModuleConfig};
 
 use crate::configs::aws::AwsConfig;
 
+fn get_aws_region_from_config(aws_profile: &Option<String>) -> Option<String> {
+    let config_location = env::var("AWS_CONFIG_FILE")
+        .ok()
+        .and_then(|path| PathBuf::from_str(&path).ok())
+        .or_else(|| {
+            let mut home = home_dir()?;
+            home.push(".aws/config");
+            Some(home)
+        })?;
+
+    let file = File::open(&config_location).ok()?;
+    let reader = BufReader::new(file);
+    let lines = reader.lines().filter_map(Result::ok);
+
+    let region_line = if let Some(ref aws_profile) = aws_profile {
+        lines
+            .skip_while(|line| line != &format!("[profile {}]", aws_profile))
+            .skip(1)
+            .take_while(|line| !line.starts_with('['))
+            .find(|line| line.starts_with("region"))
+    } else {
+        lines
+            .skip_while(|line| line != "[default]")
+            .skip(1)
+            .take_while(|line| !line.starts_with('['))
+            .find(|line| line.starts_with("region"))
+    }?;
+
+    let region = region_line.split('=').nth(1)?;
+    let region = region.trim();
+
+    Some(region.to_string())
+}
+
+fn get_aws_region() -> Option<(String, String)> {
+    env::var("AWS_DEFAULT_REGION")
+        .ok()
+        .map(|region| (String::new(), region))
+        .or_else(|| {
+            let aws_profile = env::var("AWS_PROFILE").ok();
+            let aws_region = get_aws_region_from_config(&aws_profile);
+
+            if aws_profile.is_none() && aws_region.is_none() {
```

Comment:
This whole `if` can be replaced if you use the `?` operator on lines 52 & 53:

```
let aws_profile = env::var("AWS_PROFILE").ok()?;
let aws_region = get_aws_region_from_config(&aws_profile)?;
Some((aws_profile, aws_region))
```

(Note I didn't try to compile this, but it should be fairly close)

Reply:
This is different semantic. I meant to return None when `aws_region` AND `aws_region` is both None.

Reply:
Ok, yes - I see there is a slight difference. If you think the distinction is important, then you can leave it as-is.